File: Capítulo_10/models/node.js
// Classe Node
export class Node {
    constructor(key) {
        this.key = key; // valor do nó
        this.left = null; // refêrencia ao nó filho à esquerda
        this.right = null; // refêrencia ao nó filho à direita
    }
}

// Colors
export let Colors;

(function(Colors) {
    Colors[(Colors["RED"] = 0)] = "RED"
    Colors[(Colors["BLACK"] = 1)] = "BLACK"
})(Colors || (Colors = {}));

// Classe RedBlackNode
export class RedBlackNode extends Node {
    constructor(key) {
        super(key);
        this.key = key;
        this.color = Colors.RED;
        this.parent = null;
    }
    
    isRed() {
        return this.color === Colors.RED;
    }
}

File: Capítulo_10/models/Códigos em Python/node.py
# Classe Node
class Node:
    def __init__(self, key):
        self.key = key # valor do nó
        self.left = None # refêrencia ao nó filho à esquerda
        self.right = None # refêrencia ao nó filho à esquerda

# Colors
from enum import Enum

class Colors(Enum):
    RED = 0
    BLACK = 1

# Classe RedBlackNode
class RedBlackNode(Node):
    def __init__(self, key):
        super().__init__(key)
        self.key = key
        self.color = Colors.RED
        self.parent = None

    def isRed(self):
        return self.color == Colors.RED
